public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -361,6 +365,66 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     return angular.toJson(queryHeader);
   }
 
+  showContextToggle(): boolean {
+    return this.esVersion > 5;
+  }
+
+  getLogRowContext = async (row: LogRowModel, options?: RowContextOptions): Promise<{ data: DataFrame[] }> => {
+    const sortField = row.dataFrame.fields.find(f => f.name === 'sort');
```

Comment:
That setting controls how the segments are sorted in Elasticsearch, think of it as an additional way to optimize the indices. It will speed up the sorting if those keys are the same as the keys used in your query but it doesn't have an effect on the accuracy.  It should be noted that "index sorting" does slow down indexing operations, it makes the merge operation more expensive. 

If I was going to add anything to the data source, It would be the ability to configure the "tie breaker" which is `_doc` in this PR. We've found that people rarely choose a different value because there really isn't a clear alternative that's better. That value is only needed when two log lines have the same exact timestamp.